scipy/stats/stats.py

```diff
@@ -3999,6 +3999,52 @@ def pearsonr(x, y):
     >>> stats.pearsonr([1, 2, 3, 4, 5], [10, 9, 2.5, 6, 4])
     (-0.7426106572325057, 0.1505558088534455)
 
+    It is important to keep in mind that no correlation does not imply
+    independence unless (X, Y) is Gaussian. Correlation can even be zero
+    when there is a very simple dependence structure: if X follows a
+    standard normal distribution, let Y = abs(X). A simple check
+    confirms that the correlation is close to zero:
+
+    >>> x = stats.norm.rvs(size=500)
+    >>> y = np.abs(x)
+    >>> stats.pearsonr(x, y)
+    (-0.016172891856853524, 0.7182823678751942) # may vary
+
+    Indeed, since the expectation of x is zero, cov(X, Y) = E(X*Y). This equals
+    EX*abs(X) which is zero by symmetry.
+
+    If the relationship between x and y is non-linear, the correlation
+    coefficient can be misleading. For example, if X has a standard normal
+    distribution, define Y = X if X < 0 and Y = 0 otherwise. A simple calculation
+    shows that corr(X, Y) = sqrt(2/Pi) = 0.797..., implying a high level of
+    correlation:
+
+    >>> y = np.where(x < 0, x, 0)
+    >>> stats.pearsonr(x, y)
+    (0.8537091583771509, 3.183461621422181e-143) # may vary
+
+    This is unintuitive since there is no dependence of X and Y if X is larger
+    than zero which happens in about half of the cases if we sample X and Y.
+
+    There is linear dependance between X and Y in the sense that
+    Y = a + b*X + e, were a,b are constants and e is a random error term,
```

Comment:
```suggestion
    Y = a + b*X + e, where a,b are constants and e is a random error term,
```
Do we mean there is a linear dependence between X and Y if Y = a + b*X + e, or are we now supposing that X and Y refer to random variables with this relationship?

Reply:
It is the former, I will write it more clearly